GitUI/AutoCompletion/CommitAutoCompleteProvider.cs

```diff
@@ -149,19 +149,21 @@ private static IEnumerable<string> ReadOrInitializeAutoCompleteRegexes()
 
         private static async Task<string?> GetChangedFileTextAsync(GitModule module, GitItemStatus file)
         {
-            var changes = await module.GetCurrentChangesAsync(file.Name, file.OldName, file.Staged == StagedStatus.Index, "-U1000000")
-                .ConfigureAwait(false);
-
-            if (changes is not null)
+            if (file.IsTracked)
             {
-                return changes.Text;
-            }
+                var changes = await module.GetCurrentChangesAsync(file.Name, file.OldName, file.Staged == StagedStatus.Index, "-U1000000")
+                .ConfigureAwait(false);
 
-            var content = await module.GetFileContentsAsync(file).ConfigureAwaitRunInline();
+                if (changes is not null)
+                {
+                    return changes.Text;
+                }
 
-            if (content is not null)
-            {
-                return content;
+                var content = await module.GetFileContentsAsync(file).ConfigureAwaitRunInline();
+                if (content is not null)
```

Comment:
Why?

Reply:
This is an early exit. Else it is tried another way below. So why "Why?"?

Reply:
Ok.